cmd/dra-example-controller/gpu.go

```diff
@@ -73,7 +73,7 @@ func (g *gpudriver) UnsuitableNode(crd *nascrd.NodeAllocationState, pod *corev1.
 		}
 	})
 
-	allocated := g.allocate(crd, pod, gpucas, allcas, potentialNode)
+	allocated := g.getAllocatedDevices(crd, pod, gpucas, allcas, potentialNode)
```

Comment:
What this function does, is (in code blocks) : 
1) it builds a list of what devices are available
2) removes what was already allocated previously from the freshly built list of available devices in 1)
3) check if resource claim was previously allocated and there is a record for it
3.1) if so - return the list of devices allocated previously
3.2) otherwise pick requested number of devices from available devices <- this is an actual allocation step, so the function name is not misleading, it is meant to do the allocation if it's absent.

Reply:
Thanks for the detailed explanation! I didn't understand the process inside.
I removed this commit from the PR.